src/proxy/inbound.rs

```diff
@@ -375,22 +478,82 @@ impl Inbound {
         state: &DemandProxyState,
         conn: &Connection,
         local_workload: &Workload,
-        hbone_addr: SocketAddr,
-    ) -> Result<(), Error> {
+        hbone_addr: HboneAddress,
+    ) -> Result<Option<SocketAddr>, Error> {
         let illegal_call = cfg.illegal_ports.contains(&hbone_addr.port());
         if illegal_call {
             return Err(Error::SelfCall);
         }
 
-        if conn.dst.ip() == hbone_addr.ip() {
-            // Normal case: both are aligned. This is allowed (we really only need the HBONE address for the port.)
-            return Ok(());
+        match hbone_addr {
+            // TODO(jaellio): Should we also validate the hbone_addr port?
+            HboneAddress::SvcHostname(hbone_host, hbone_port) => {
+                // Validate a service or workload exists for the hostname
+                let parts: Vec<&str> = hbone_host.split('.').collect();
```

Comment:
TODO - figure out if there is a better way to parse a hostname

Reply:
Took a shot at this with ranking - might be a simpler way to do it in rust but I am not at all familiar :) Let me know if it's similar to what you had in mind.